client/src/App.jsx

```diff
@@ -5,19 +5,22 @@ import MainPage from './components/MainPage';
 import Header from './components/common/Header';
 import Footer from './components/common/Footer';
 import ProfilePage from './components/ProfilePage';
+import { ModalProvider } from './components/HomePage/modal/ModalContext';
 
 export default function App() {
   return (
     <Router>
-      <Fragment>
-        <Header />
-        <Switch>
-          <Route path="/landing" component={HomePage} />
-          <Route path="/profile" component={ProfilePage} />
-          <Route path="/" component={MainPage} />
-        </Switch>
-        <Footer />
-      </Fragment>
+      <ModalProvider>
+        <Fragment>
+          <Header />
+          <Switch>
+            <Route path="/landing" component={HomePage} />
```

Comment:
any incorrect path will redirect to the main page like this.
```suggestion
            <Route exact path="/landing" component={HomePage} />
```